### PR TITLE
Adding `allsigs`, the dependent version of `allpairs`

### PR DESCRIPTION
**Pull request by CohenCyril**

### CODE REVIEW

File: ChangeLog [hunk 1]
```diff
@@ -1,6 +1,8 @@
 ??/??/???? - version 1.7.1
 	* Added companion matrix of a polynomial `companionmx p` and the
 	  theorems: companionmxK, map_mx_companion and companion_map_poly
+	* Added allsigs, the dependent version of allpairs,
+	  with notation `[seq E | i <- s & j <- t i]`
```
Comment (thery) [inline ChangeLog]: `[seq E | i <- s & j <- t]`

Comment (CohenCyril) [reply ChangeLog]: The phrasing `[seq E | i <- s & j <- t i]` has the advantage to explicit the dependency of `i` in `t`, do you think it is better kept implicit in the doc?

Comment (thery) [reply ChangeLog]: From finset
[set E | x in A, y in B] == the set of values of E for x drawn from A and *)
(*                      and y drawn from B; B may depend on x.

Comment (CohenCyril) [reply ChangeLog]: @thery ok, fixed

**Comment by thery** [comment]
merging before the meeting

**Pull request merged**